File: samples/py/stack_test.py
cur_thd = dbgscript.current_thread()

print ("Engine Thd ID:", cur_thd.engine_id, "Sys Id:", cur_thd.thread_id)

stack = cur_thd.get_stack()

frame_idx = 2

locals = stack[frame_idx].get_locals()
args = stack[frame_idx].get_args()

print ("Locals:")
for v in locals:
	print ("Name:", v.name, "Size:", v.size, "Type:", v.type, "Module:",
        v.module, "Addr:", v.address)
	
print ("Args:")
for v in args:
	print ("Name:", v.name, "Size:", v.size)

File: samples/rb/stack_test.rb
thread = DbgScript.get_threads[0]
cur_thd = DbgScript.current_thread

stack = thread.get_stack
puts thread.teb.to_s(16)

for f in stack do
    locals = f.get_locals

    puts "Locals:"
    for v in locals
        puts \
            "Name: #{v.name}, "  \
            "Type: #{v.type}, "  \
            "Mod: #{v.module}, " \
            "Size: #{v.data_size}, " \
            "Addr: #{v.address.to_s(16)}"
    end

end

